device/samples/javascript/device_through_proxy.js

```diff
@@ -23,7 +23,10 @@ const deviceConnectionString = process.env.DEVICE_CONNECTION_STRING;
 let sendInterval;
 
 function disconnectHandler () {
-  clearInterval(sendInterval);
+  if (!!sendInterval) {
```

Comment:
Based on my testing, `clearInterval(null)` and `clearInterval(undefined)` doesn't cause an error, so this `if` could potentially be removed entirely. Same feedback applies to the other samples.